src/pid.cpp

```diff
@@ -81,16 +81,28 @@ Pid::Pid(const Pid & source)
 Pid::~Pid() {}
 
 void Pid::initPid(
-  double p, double i, double d, double i_max, double i_min, NodeParamsIfacePtr node_param_iface)
+  double p, double i, double d, double i_max, double i_min, NodeParamsIfacePtr node_param_iface,
+  std::string parameter_prefix)
 {
   const Pid::Gains gains = getGains();
-  initPid(p, i, d, i_max, i_min, gains.antiwindup_, node_param_iface);
+  initPid(p, i, d, i_max, i_min, gains.antiwindup_, node_param_iface, parameter_prefix);
+}
+
+bool endsWith(const std::string & str, const std::string & suffix)
+{
+  return str.size() >= suffix.size() &&
+         0 == str.compare(str.size() - suffix.size(), suffix.size(), suffix);
 }
 
 void Pid::initPid(
   double p, double i, double d, double i_max, double i_min, bool antiwindup,
-  NodeParamsIfacePtr node_param_iface)
+  NodeParamsIfacePtr node_param_iface, std::string parameter_prefix)
+
 {
+  parameter_prefix_ = parameter_prefix;
+  if (!parameter_prefix_.empty() && !endsWith(parameter_prefix_, ".")) {
```

Comment:
I don't think it needs a dedicated function for this (`endsWith`). Why not just checking for `*parameter_prefix_.end() != '.'`?

Reply:
Fixed https://github.com/ros-controls/control_toolbox/pull/94/commits/14f52e3bca5180c19a2a653433d742b0516adc8c